File: Examples/Objective-C/Remote Control/Remote Control/ImagePickerViewController.m
//
//  Copyright © 2020 Bridgefy Inc. All rights reserved.
//

#import "ImagePickerViewController.h"
#import "AdminViewTabBarController.h"

@interface ImagePickerViewController ()

@property (nonatomic, retain) UIButton *currentButton;

@end

@implementation ImagePickerViewController

- (IBAction)imageButtonPressed:(id)sender {
    UIButton *selectedButton = (UIButton *)sender;
    
    [self.currentButton setImage:[UIImage imageNamed:@"radioButtonOff"]
                        forState:UIControlStateNormal];
    
    [selectedButton setImage:[UIImage imageNamed:@"radioButtonOn"]
                    forState:UIControlStateNormal];
    
    self.currentButton = selectedButton;
    
    if (self.tabBarController) {
        AdminViewTabBarController *tabBarController = (AdminViewTabBarController *)self.tabBarController;
        [tabBarController sendObject:@(selectedButton.tag) sender:self];
    }
}


@end

File: Examples/Swift/Remote Control/Remote Control/ImagePickerViewController.swift
//
//  Copyright © 2020 Bridgefy Inc. All rights reserved.
//

import UIKit

class ImagePickerViewController: UIViewController {
    
    var currentButton: UIButton?
    
    @IBAction func imageButtonPressed(_ sender: Any) {
        let selectedButton = sender as! UIButton
        
        currentButton?.setImage(#imageLiteral(resourceName: "radioButtonOff"), for: .normal)
        selectedButton.setImage(#imageLiteral(resourceName: "radioButtonOn"), for: .normal)
        currentButton = selectedButton
        
        if let tabBarController = tabBarController as? AdminViewTabBarController {
            tabBarController.send(object: selectedButton.tag as Any, sender: self)
        }
    }
    
}
